### DISCUSSION TITLE
iotdb配置C++原生接口

### DISCUSSION THREAD

**Discussion by Pummole** [apache/iotdb#12462] [Q&A]
根据iotdb用户手册，在根目录下执行mvn package -P compile-cpp -pl example/client-cpp-example -am -DskipTest，报错
![11](https://github.com/apache/iotdb/assets/99628201/a2cfd7d9-b578-466a-bd95-0c69c4c7a3f2)

**Comment by HTHou** [comment]
用户手册有点问题，可以尝试把 -P compile-cpp 替换为 -P with-cpp 看看

**Comment by Pummole** [comment]
多谢啦，已经解决了，根据github中的描述，修改了client-cpp下的pom.xml，目前已经构建出了压缩包，里面含有Session.h等头文件

---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2024年05月22日 15:21 |
| 收件人 | apache/iotdb ***@***.***> |
| 抄送人 | Pummole ***@***.***>,
Author ***@***.***> |
| 主题 | Re: [apache/iotdb] iotdb配置C++原生接口 (Discussion #12462) |

用户手册有点问题，可以尝试把 -P compile-cpp 替换为 -P with-cpp 看看

**Comment by GentleQuite** [comment]
大佬，是怎么改的，可以讲一下位置吗

**Reply by Pummole** [reply]
修改了client-cpp下的pom.xml ；把里面的test那一段删掉试试，然后换了一个下载源，也在pom里面换的

**Reply by GentleQuite** [reply]
能加个好友（Q2623582019）吗，不太会看pom文件

**Comment by spacefang** [comment]
花了一下午时间终于编译成功了，我手动下载了apache-maven-3.9.6-bin.zip，maven-wrapper-3.3.2.jar，以及iotdb-bin-resources-main.zip文件，然后进行手动编译thrift,同时我在编译命令中手动指定 Boost 库的路径，直接告诉 cmake 去哪里找它需要的文件。
这是我的编译命令，希望能给遇到此问题提供借鉴：
`./mvnw clean package \
-pl example/client-cpp-example \
-am \
-DskipTests \
-P with-cpp \
-Diotdb-tools-thrift.version=0.20.0.0 \
-Dcmake.args="-DboostLibraryDir=/usr/lib/x86_64-linux-gnu/"`
<img width="1333" height="437" alt="image" src="https://github.com/user-attachments/assets/01b97461-1413-4831-b75e-1809c5f53af8" />
